Bug: Latest release of `@xstate/store` has broken exports

### XState version

XState version 5

### Description

The latest published release (v3.17.4) of `@xstate/store` has broken exports. The install size is also significantly smaller than before (22KB vs 180KB)

I presume something with the build process went differently than it did in v3.17.3. Maybe related to #5517?

Best I can tell it's because the publish job in the updated workflow doesn't have:
```
- uses: ./.github/actions/ci-checks
```

which runs the build process.

### Expected result

Importing from `@xstate/store` works

### Actual result

With the code:
```js
import { createStore } from "@xstate/store"
```

I get the following error when running directly in node:
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/path/to/project/node_modules/@xstate/store/dist/dist/xstate-store.cjs.js' imported from /path/to/project/node_modules/@xstate/store/dist/xstate-store.cjs.mjs
```

With `vite dev` you'll get a invalid imports/exports error.


### Reproduction

https://stackblitz.com/edit/github-jaznwfud?file=src%2Fmain.ts

### Additional context

The `xstate-store.cjs.js` file looks like it was published with dev content:
```
/// ...
/// some code I've clipped from the snippet
/// ...

// this re-exports the source file
module.exports = require("../src/index.ts");

unregister();
````

--- a/.changeset/fix-publish-build.md
+++ b/.changeset/fix-publish-build.md
@@ -1,0 +1,5 @@
+---
+'@xstate/store': patch
+---
+
+Fix broken exports caused by missing build step in the publish workflow.

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -47,6 +47,9 @@ jobs:
       - uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
       - uses: ./.github/actions/ci-setup
 
+      - name: Build
+        run: pnpm build
+
       - name: Publish to npm
         # https://github.com/changesets/action
         uses: changesets/action@6a0a831ff30acef54f2c6aa1cbbc1096b066edaf # v1.7.0

